src/seraphis_main/enote_finding_context.h

```diff
@@ -84,4 +89,131 @@ class EnoteFindingContextLedger
         const std::uint64_t chunk_max_size) const = 0;
 };
 
+////
+/// LegacyUnscannedTransaction: a transaction that is ready to be legacy view scanned
+///
+struct LegacyUnscannedTransaction final
+{
+    rct::key transaction_id;
+    uint64_t unlock_time;
+    sp::TxExtra tx_memo;
+    uint64_t total_enotes_before_tx;
+    std::vector<sp::LegacyEnoteVariant> enotes;
+    std::vector<crypto::key_image> legacy_key_images;
+};
+
+////
+/// LegacyUnscannedBlock: a block that is ready to be legacy view scanned
+// - the txs are expected to be ordered as they appear in the block, where
+// the first tx is the miner tx
+///
+struct LegacyUnscannedBlock final
+{
+    uint64_t block_index;
+    uint64_t block_timestamp;
+    rct::key block_hash;
+    rct::key prev_block_hash;
+    std::vector<LegacyUnscannedTransaction> unscanned_txs;
+};
+
+////
+/// LegacyUnscannedChunk: a chunk of blocks ready to be legacy view scanned
+// - the blocks are expected to match their order on chain
+///
+typedef std::vector<LegacyUnscannedBlock> LegacyUnscannedChunk;
```

Comment:
```suggestion
using LegacyUnscannedChunk = std::vector<LegacyUnscannedBlock>;
```
Stylistic consistency.

Using statements are more clear IMO.